client/src/components/Navbar.js

```diff
@@ -2,15 +2,51 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { Nav, Navbar, NavItem, NavDropdown, MenuItem } from 'react-bootstrap';
 import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import { logSignerOut } from '../redux/actions/signature';
 
-const NavBar = props => {
-  let homeText;
-  if (props.location.pathname === '/') {
-    homeText = 'RE:IMAGINE DENVER';
-  } else {
-    homeText = 'HOME';
+// RENDERS MyCampaignNavItem BASED ON AUTH STATUS
+function MyCampaignNavItem(props) {
+  return (
+    // if logged in and signed a campaign, show MyCampaign
+    <NavItem eventKey={4}>
+      <Link to="">My Campaign</Link>
+    </NavItem>
+  );
+}
+
+// RENDERS UserAuthNav ITEM WITH LOGIN / LOGOUT BASED ON AUTH STATUS
+function UserAuthNav(props) {
+  // checks if user is not currently authed with google or facebook
+  if (props.auth && (!props.auth.googleID && !props.auth.facebookID)) {
+    return (
+      // if not logged in > show sign in options
+      <NavDropdown id="tools-dropdown" eventKey={5} title="Login">
+        <MenuItem eventKey={5.1} href="/auth/facebook">
+          Sign in With Facebook
+        </MenuItem>
+        <MenuItem eventKey={5.2} href="/auth/google">
+          Sign in With Google
+        </MenuItem>
+      </NavDropdown>
+    );
   }
+  const { auth: { name } } = props;
+  const firstName = name.substr(0, name.indexOf(' '));
+  const logOutUser = props.logOutUser;
+  // if logged in > show sign out options
+  return (
+    <NavItem eventKey={5} onClick={logOutUser}>
+      Sign Out, {firstName}
+    </NavItem>
+  );
+}
 
+const NavBar = props => {
+  // used "!!" to prevent undefined. In brief !undefined = true > !true = false
```

Comment:
// used "!!" to cast variable to boolean.  Used to avoid the behavior of `undefined` in conditional logic. In brief !undefined = true > !true = false